src/xpcc/communication/xpcc/backend/zeromq/connector.hpp

```diff
@@ -0,0 +1,188 @@
+// coding: utf-8
+/* Copyright (c) 2016, Roboterclub Aachen e. V.
+ * All Rights Reserved.
+ *
+ * The file is part of the xpcc library and is released under the 3-clause BSD
+ * license. See the file `LICENSE` for the full license governing this code.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef	XPCC__ZEROMQ_CONNECTOR_HPP
+#define	XPCC__ZEROMQ_CONNECTOR_HPP
+
+#include <cstring>		// for std::memcpy
+
+#include <zmqpp/zmqpp.hpp>
+
+#include <xpcc/debug/logger.hpp>
+#undef XPCC_LOG_LEVEL
+#define	XPCC_LOG_LEVEL xpcc::log::ERROR
+
+#include "../backend_interface.hpp"
+
+namespace xpcc
+{
+class ZeroMQConnectorBase
+{
+public:
+	enum class Mode
+	{
+		SubPush, /// In this mode the backend connects to a remote machine.
+		PubPull, /// Server mode in which the backend binds to two ports. The ports must be accessible.
+	};
+};
+
+class ZeroMQConnector : public ZeroMQConnectorBase, public BackendInterface
+{
+public:
+	ZeroMQConnector(std::string endpointIn, std::string endpointOut, Mode mode = Mode::SubPush) :
+		socketIn (context, (mode == Mode::SubPush ? zmqpp::socket_type::sub  : zmqpp::socket_type::pull)),
+		socketOut(context, (mode == Mode::SubPush ? zmqpp::socket_type::push : zmqpp::socket_type::pub)),
+		isAvailable(false)
+	{
+		switch(mode)
+		{
+			case
+			Mode::SubPush:
+				this->socketIn.connect(endpointIn);
+				this->socketIn.subscribe("");
+
+				this->socketOut.connect(endpointOut);
+			break;
+
+			case
+			Mode::PubPull:
+				this->socketIn.bind(endpointIn);
+				this->socketOut.bind(endpointOut);
+			break;
+		}
+	}
+
+	virtual
+	~ZeroMQConnector() {}
```

Comment:
Probably the destructor will take care of it. I added `unsubscribe()` and `close()` where possible.